digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/ServicePointBusStopDao.scala

```diff
@@ -0,0 +1,262 @@
+        from asset a
+          join property p on a.asset_type_id = p.asset_type_id
+          left join single_choice_value s on s.asset_id = a.id and s.property_id = p.id and p.property_type = 'single_choice'
+          left join text_property_value tp on tp.asset_id = a.id and tp.property_id = p.id and (p.property_type = 'text' or p.property_type = 'long_text' or p.property_type = 'read_only_text')
+          left join multiple_choice_value mc on mc.asset_id = a.id and mc.property_id = p.id and (p.property_type = 'multiple_choice' or p.property_type = 'checkbox')
+          left join enumerated_value e on mc.enumerated_value_id = e.id or s.enumerated_value_id = e.id
+        where exists (
+            select '1'
+            from multiple_choice_value aux_mc
+            join enumerated_value aux_e on aux_mc.enumerated_value_id = aux_e.id and aux_e.value = '7'
+            where aux_mc.asset_id = a.id) and (a.valid_to > sysdate or a.valid_to is null)
+      """
+    queryToServicePoint(queryFilter(query))
+  }
+
+  private def queryToServicePoint(query: String): Seq[ServicePoint] = {
+    val rows = Q.queryNA[ServicePointRow](query).iterator.toSeq
+
+    rows.groupBy(_.id).map { case (id, stopRows) =>
+      val row = stopRows.head
+      val commonProperties: Seq[Property] = AssetPropertyConfiguration.assetRowToCommonProperties(row)
+      val properties: Seq[Property] = commonProperties ++ assetRowToProperty(stopRows)
+      val point = row.point.get
+      val stopTypes = extractStopTypes(stopRows)
+
+      id -> ServicePoint(id = row.id, nationalId = row.externalId, stopTypes = stopTypes,
+        municipalityCode = row.municipalityCode, lon = point.x, lat = point.y, created = row.created, modified = row.modified, propertyData = properties)
+    }.values.toSeq
+  }
+
+  private def extractStopTypes(rows: Seq[ServicePointRow]): Seq[Int] = {
+    rows.filter(_.property.publicId.equals(MassTransitStopOperations.MassTransitStopTypePublicId)).map { row => row.property.propertyValue.toInt }
+  }
+
+  def expire(id: Long, username: String) = {
+    Queries.updateAssetModified(id, username).execute
+    sqlu"update asset set valid_to = sysdate where id = $id".execute
+  }
+
+  def update(assetId: Long, properties: Seq[SimplePointAssetProperty], user: String) = {
+    sqlu"""
+           UPDATE asset
+            SET modified_by = $user, modified_date = sysdate
+            WHERE id = $assetId""".execute
+
+    updateAssetProperties(assetId, properties)
+    assetId
+  }
+
+  def insertAsset(id: Long, lon: Double, lat: Double, creator: String, municipalityCode: Int): Unit = {
+    sqlu"""insert into asset (id, external_id, asset_type_id, created_by, municipality_code, geometry)
+           select $id, national_bus_stop_id_seq.nextval, $typeId, $creator, $municipalityCode,
+           MDSYS.SDO_GEOMETRY(4401, 3067, NULL, MDSYS.SDO_ELEM_INFO_ARRAY(1,1,1), MDSYS.SDO_ORDINATE_ARRAY($lon, $lat, 0, 0))
+           from dual
+      """.execute
+  }
```

Comment:
It's missing floating value even if always false

Reply:
Does it matter?

Reply:
@marquesfm Nothing done here. Leaved it as it is